src/VisualStudio/Core/Test/AbstractTextViewFilterTests.vb

```diff
@@ -45,34 +45,189 @@ Namespace Microsoft.VisualStudio.LanguageServices.UnitTests
+                            {
+                                Console.WriteLine(Increment[|$$(5)|]);
+                            }
+                            static int Increment(int n)
+                            {
+                                return n+1;
+                            }
+                        }
+                        </Document>
+                    </Project>
+                </Workspace>
+
+            Using workspace = TestWorkspaceFactory.CreateWorkspace(workspaceXml)
+                Dim doc = workspace.Documents.Single()
+                Dim span = doc.SelectedSpans.Single()
+                TestSpan(workspace, doc, span.Start, span.End, commandId:=CUInt(VSConstants.VSStd2KCmdID.GOTOBRACE))
+                TestSpan(workspace, doc, span.End, span.Start, commandId:=CUInt(VSConstants.VSStd2KCmdID.GOTOBRACE))
+                TestSpan(workspace, doc, span.Start + 1, span.End, commandId:=CUInt(VSConstants.VSStd2KCmdID.GOTOBRACE))
+                TestSpan(workspace, doc, span.End - 1, span.Start, commandId:=CUInt(VSConstants.VSStd2KCmdID.GOTOBRACE))
+            End Using
+        End Sub
+
+        <Fact, Trait(Traits.Feature, Traits.Features.BraceMatching)>
+        Public Sub GotoBraceExtFindsTheInnerPositionOfCloseBraceAndOuterPositionOfOpenBrace()
+            Dim workspaceXml =
+                <Workspace>
+                    <Project Language=<%= LanguageNames.CSharp %> CommonReferences="true">
+                        <Document>
+                        using System;
+                        class C
+                        {
+                            static void M()
+                            {
+                                Console.WriteLine[|(Increment(5)|])$$;
+                            }
+                            static int Increment(int n)
+                            {
+                                return n+1;
+                            }
+                        }
+                        </Document>
+                    </Project>
+                </Workspace>
+
+            Using workspace = TestWorkspaceFactory.CreateWorkspace(workspaceXml)
+                Dim doc = workspace.Documents.Single()
+                Dim span = doc.SelectedSpans.Single()
+                TestSpan(workspace, doc, caretPosition:=span.Start, startPosition:=span.Start, endPosition:=span.End, commandId:=CUInt(VSConstants.VSStd2KCmdID.GOTOBRACE_EXT))
+                TestSpan(workspace, doc, caretPosition:=doc.CursorPosition.Value, startPosition:=span.End, endPosition:=span.Start, commandId:=CUInt(VSConstants.VSStd2KCmdID.GOTOBRACE_EXT))
+            End Using
+        End Sub
+
+        <Fact, Trait(Traits.Feature, Traits.Features.BraceMatching)>
+        Public Sub GotoBraceExtFromLeftAndRightOfOpenAndCloseBraces()
+            Dim workspaceXml =
+                <Workspace>
+                    <Project Language=<%= LanguageNames.CSharp %> CommonReferences="true">
```

Comment:
Yes, VB works. I didn't add specific tests for VB as it flows through the same code and its language agnostic up here. Now I'm thinking that its still a good idea to add some VB tests to document the fact that it works..